### DISCUSSION TITLE
Question about Hill Plasticity Stress Update

### DISCUSSION THREAD

**Discussion by WildBobby** [idaholab/moose#19318] [Q&A Modules: Solid mechanics] [answered]
Hi, I am a student working on a project about anisotropic material. I am very new to plasticity and MOOSE, and I need help to better understand the "[Hill Plasticity Stress Update](https://mooseframework.inl.gov/source/materials/ADHillPlasticityStressUpdate.html)".

In the given example 
(modules/tensor_mechanics/test/tests/ad_anisotropic_plasticity/ad_aniso_plasticity_x.i)
the given yield stress is 0.001, while in the given result
(modules/tensor_mechanics/test/tests/ad_anisotropic_plasticity/gold/ad_aniso_plasticity_x_out.csv)
the initial yield stress is 0.001417, (about sqrt(2) times of the given yield stress).

I did not make any revision of the input file, and was expecting the initial yield stress the same as 0.001. I am wondering, do I have any misunderstanding of what the input file is doing here, or is there any rotation which could scale the result by sqrt(2) while I did not notice?

Any suggestion or comment would be appreciated! Many thanks!

**Comment by recuero** [comment, marked as answer]
When you have anisotropic plasticity, the stress components contribute to the yield criterion through some sort of anisotropic coefficients. In this case, you have the Hill's coefficients (F, G, H, L, M, N) modifying the current stress state at the material point to quantify its contribution to the total stress. So, even though you may select 0.001 as the yield stress, that doesn't really mean your uniaxial stress will reach that value at the beginning of plastic yield. However, it must be on the order of magnitude, since the anisotropy coefficients usually don't go beyond a factor of 2 or 3.

From https://mooseframework.inl.gov/source/materials/ADHillPlasticityStressUpdate.html, if you develop that equation and substitute the coefficients (F, G, H, L, M, N)  in the Hill's tensor ("A" on that doc page), you can arrive at an equation that will give you a number for sigma_x in that test that you point to. Hopefully, that number is in agreement with the gold file.

**Reply by WildBobby** [reply]
Thank you for answering my question. Sorry that I previously did not explain things clearly. I agree that the value of Hill's coefficients would influence the stress components, and according to [the definition of Hill's coefficient ](https://abaqus-docs.mit.edu/2017/English/SIMACAEMATRefMap/simamat-c-anisoyield.htm) (I hope that you do not mind me using Abaqus' explanation), I know that
F = (sigma_0)^2/2 * (1/(sigma_2)^2 + 1/(sigma_3)^2 - 1/(sigma_1)^2)
G = (sigma_0)^2/2 * (1/(sigma_3)^2 + 1/(sigma_1)^2 - 1/(sigma_2)^2)
and
H = (sigma_0)^2/2 * (1/(sigma_1)^2 + 1/(sigma_2)^2 - 1/(sigma_3)^2). 
where sigma_0 is the initial stress, sigma_1, sigma_2 and sigma_3 are the normal yield stresses.

Then the relation between the yield stress in x direction would be
G + H = (sigma_0)^2/2 * (2/(sigma_1)^2)=(sigma_0/sigma_1)^2
or sigma_1 = sigma_0 * sqrt(1/(G+H)).
I think this derivation also matches what is given in MOOSE's definition of Hill's tensor, as A(1,1)=G+H.

In the given example (modules/tensor_mechanics/test/tests/ad_anisotropic_plasticity/ad_aniso_plasticity_x.i)
F = 0.5829856, G = 0.364424, H = 0.6342174, and sigma_0 ( yield stress) = 0.001
Then G+H=0.9986414, sigma_1 = sigma_0 * sqrt(1/(G+H))=0.001 * sqrt(1/0.9986414) which approximate to 0.001. That is why I expected the initial yield stress would be 0.001 instead of 0.0014 (the result from the given gold file)

So I was thinking, did I make any mistake on the above derivation or did I miss any important component for the calculation?

Thank you again for answering my question!

**Reply by recuero** [reply]
No, you didn't miss anything. There is a factor of 2 (actually a divide-by-2) that we are applying to the current stress contribution (see our doc page), so the definition is slightly different from the page you just pointed to (this explains the math). I think I remember seeing this type of discrepancies out in the literature. Hopefully, it's not too big of a hassle.

**Reply by WildBobby** [reply]
I see what you mean now. This makes a lot of sense. Thank you for the help!
